### PR TITLE
Improving usability

### PR DESCRIPTION
**Pull request by WebVPF**
Improving usability - backend section

### CODE REVIEW

File: ajax-javascript-api.md [hunk 1]
```diff
@@ -48,56 +62,76 @@ Handler | Description
...
 <a name="global-events"></a>
 ## Global AJAX events
 
 The AJAX framework triggers several events on the updated elements, triggering element, form, and the window object. The events are triggered regardless on which API was used - the data attributes API or the JavaScript API.
 
+<style>
+    .attributes-table-precessor + table td:first-child,
+    .attributes-table-precessor + table td:first-child > * { white-space: nowrap; }
+</style>
+<div class="attributes-table-precessor"></div>
```
Comment (bennothommo) [inline ajax-javascript-api.md:133]: @WebVPF what is the purpose of this inline styling?

File: ajax-update-partials.md [hunk 2]
```diff
@@ -43,35 +47,42 @@ The definition of what should be updated is specified as a JSON-like object wher
 
 The following will request to update the `#myDiv` element with **mypartial** contents.
 
-    mypartial: '#myDiv'
+```none
```
Comment (bennothommo) [inline ajax-update-partials.md:50]: @WebVPF for codeblocks with `none`, could you just remove the `none` so they are interpreted as raw text?

File: backend-reorder.md [hunk 3]
```diff
@@ -20,58 +20,63 @@ The behavior depends on a [model class](../database/model) which must implement
...
-	toolbar:
-	    # Partial for toolbar buttons
-	    buttons: reorder_toolbar
+# Model Class name
+modelClass: Acme\Shop\Models\Category
 
+# Toolbar widget configuration
+toolbar:
+	# Partial for toolbar buttons
+	buttons: reorder_toolbar
```
Comment (bennothommo) [inline backend-reorder.md:60]: @WebVPF spaces only for whitespace, please :)

```suggestion
    # Partial for toolbar buttons
    buttons: reorder_toolbar
```

**Comment by bennothommo** [comment]
@WebVPF great work with this. I've just left a couple of fixes for you to do.

File: ajax-javascript-api.md (hunk 1 above)
Comment (WebVPF) [reply ajax-javascript-api.md:133]: no styles:



with styles:



the solution is taken here (before table): https://github.com/wintercms/docs/blob/main/ajax-attributes-api.md

File: ajax-update-partials.md (hunk 2 above)
Comment (WebVPF) [reply ajax-update-partials.md:50]: @bennothommo If you use modern libraries to highlight the code, the `none` indicator will style the code block but will not highlight the syntax of the code.
Perhaps in the future you will want to use the PrismJS (https://prismjs.com) library on the documentation site.
Without the indicator, the code block will not be styled at all.

Maybe it's better to leave `none`?

**Comment by WebVPF** [comment]
I will continue working when this pull request is accepted.

File: ajax-update-partials.md (hunk 2 above)
Comment (LukeTowers) [reply ajax-update-partials.md:50]: would this not just be `json`?

Comment (LukeTowers) [reply ajax-update-partials.md:50]: @WebVPF if you would be willing to just change these examples to be in the form of:

```js
$.request('onExample', { update: {
    mypartial: '#myDiv'
}});
```
instead so we can use JS highlighting that would be great

File: backend-controllers-ajax.md [hunk 4]
```diff
@@ -53,67 +55,83 @@ The backend controller base class defines a number of properties that allow to c
...
+`$guarded` | controller specific methods which cannot be called as actions. Can be extended in the controller constructor.
+`$layout` | specify a custom layout for the controller views (see [layouts](#layouts) below).
 
 <a name="actions-views-routing"></a>
 ## Actions, views and routing
 
 Public controller methods, called **actions** are coupled to **view files** which represent the page corresponding the action. Backend view files use PHP syntax. Example of the **index.htm** view file contents, corresponding to the **index** action method:
 
-    <h1>Hello World</h1>
+```php
```
Comment (LukeTowers) [inline backend-controllers-ajax.md:75]:
```suggestion
```html
```

File: backend-reorder.md [hunk 5]
```diff
@@ -20,58 +20,63 @@ The behavior depends on a [model class](../database/model) which must implement
...
+		'Backend.Behaviors.ReorderController',
+	];
 
-        public $reorderConfig = 'config_reorder.yaml';
+	public $reorderConfig = 'config_reorder.yaml';
 
-        // [...]
-    }
+	// [...]
+}
```
Comment (LukeTowers) [inline backend-reorder.md:35]: inner indentation needs to be fixed here

File: backend-reorder.md [hunk 6]
```diff
@@ -92,7 +97,9 @@ in
...
 The lookup query for the list [database model](../database/model) can be extended by overriding the `reorderExtendQuery` method inside the controller class. This example will ensure that soft deleted records are included in the list data, by applying the **withTrashed** scope to the query:
 
-	public function reorderExtendQuery($query)
-	{
-	    $query->withTrashed();
-	}
+```php
+public function reorderExtendQuery($query)
+{
+	$query->withTrashed();
```
Comment (LukeTowers) [inline backend-reorder.md:103]:
```suggestion
    $query->withTrashed();
```

File: backend-users.md [hunk 7]
```diff
@@ -128,25 +146,29 @@ The backend user model has methods that allow to determine whether the user has
...
-    <?php if ($this->user->hasAccess('acme.blog.delete_categories')): ?>
-        <button
-            type="button"
-            class=wn-icon-trash-o btn-icon danger pull-right"
-            data-request="onDelete"
-            data-load-indicator="Deleting Category..."
-            data-request-confirm="Do you really want to delete this category?">
-        </button>
-    <?php endif ?>
+```html
```
Comment (LukeTowers) [inline backend-users.md:164]:
```suggestion
```php
```

File: ajax-update-partials.md (hunk 2 above)
Comment (WebVPF) [reply ajax-update-partials.md:50]: @LukeTowers No. Will compile without language indicator.

without `none`:

```html
<pre>
    <code></code>
</pre>
```

with `none`:

```html
<pre class="language-none">
    <code></code>
</pre>
```

This is used to style a block of code. This is how they define it in CSS:

```css
pre[class*="language-"] {
    background: #f5f2f0;
}
```

if necessary I can delete all `none`

Comment (LukeTowers) [reply ajax-update-partials.md:50]: @bennothommo I think I'm fine with the none indicator then, any further thoughts?

Comment (bennothommo) [reply ajax-update-partials.md:50]: I'd rather not include it. We're styling non-defined code blocks the same as defined code blocks in the docs, so it doesn't matter.

File: backend-reorder.md (hunk 5 above)
Comment (WebVPF) [reply backend-reorder.md:35]: @LukeTowers How to fix it?

Comment (WebVPF) [reply backend-reorder.md:35]: Like this?

```
// [...]
}
```

**Comment by WebVPF** [comment]
Remove `none`?
What to use where there is no language indicator? Indentation or three apostrophes?

**Comment by bennothommo** [comment]
@WebVPF let's use three apostrophes as the standard from now on. In the case of `none`, simply change it so it's no language at all.

**Comment by WebVPF** [comment]
It is bad without `none` because it is not clear where the beginning of the code is.

File: backend-lists.md [hunk 8]
```diff
@@ -46,85 +48,93 @@ The list behavior depends on list [column definitions](#list-columns) and a [mod
...
+`searchOnEnter` | setting this to true will make the search widget wait for the Enter key to be pressed before it starts searching (the default behavior is that it starts searching automatically after someone enters something into the search field and then pauses for a short moment).  Default: `false`.
 
 The toolbar buttons partial referred above should contain the toolbar control definition with some buttons. The partial could also contain a [scoreboard control](../ui/scoreboard) with charts. Example of a toolbar partial with the **New Post** button referring to the **create** action provided by the [form behavior](forms):
 
-    <div data-control="toolbar">
-        <a
-            href="<?= Backend::url('acme/blog/posts/create') ?>"
-            class="btn btn-primary wn-icon-plus">New Post</a>
-    </div>
+```html
```
Comment (LukeTowers) [inline backend-lists.md:122]:
```suggestion
```php
```

File: database-query.md [hunk 9]
```diff
@@ -219,77 +262,91 @@ The `groupByRaw` method may be used to set a raw string as the value of the grou
...
 
 To perform a "cross join" use the `crossJoin` method with the name of the table you wish to cross join to. Cross joins generate a cartesian product between the first table and the joined table:
 
-    $users = Db::table('sizes')
-                ->crossJoin('colors')
-                ->get();
+```php
+$users = Db::table('sizes')
+            ->crossJoin('colors')
+            ->get();
```
Comment (LukeTowers) [inline database-query.md:294]:
```suggestion
    ->crossJoin('colors')
    ->get();
```

File: database-query.md [hunk 10]
```diff
@@ -219,77 +262,91 @@ The `groupByRaw` method may be used to set a raw string as the value of the grou
...
 
-    $latestPosts = Db::table('posts')
-                       ->select('user_id', Db::raw('MAX(created_at) as last_post_created_at'))
-                       ->where('is_published', true)
-                       ->groupBy('user_id');
+```php
+$latestPosts = Db::table('posts')
+                    ->select('user_id', Db::raw('MAX(created_at) as last_post_created_at'))
+                    ->where('is_published', true)
+                    ->groupBy('user_id');
```
Comment (LukeTowers) [inline database-query.md:328]:
```suggestion
    ->select('user_id', Db::raw('MAX(created_at) as last_post_created_at'))
    ->where('is_published', true)
    ->groupBy('user_id');
```

File: database-query.md [hunk 11]
```diff
@@ -219,77 +262,91 @@ The `groupByRaw` method may be used to set a raw string as the value of the grou
...
 
-    $users = Db::table('users')
-                ->joinSub($latestPosts, 'latest_posts', function ($join) {
-                    $join->on('users.id', '=', 'latest_posts.user_id');
-                })->get();
+$users = Db::table('users')
+            ->joinSub($latestPosts, 'latest_posts', function ($join) {
+                $join->on('users.id', '=', 'latest_posts.user_id');
+            })->get();
```
Comment (LukeTowers) [inline database-query.md:333]:
```suggestion
    ->joinSub($latestPosts, 'latest_posts', function ($join) {
        $join->on('users.id', '=', 'latest_posts.user_id');
    })->get();
```

File: database-query.md [hunk 12]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
 
 Winter CMS also supports querying JSON column types on databases that provide support for JSON column types. To query a JSON column, use the `->` operator:
 
-    $users = Db::table('users')
-                    ->where('options->language', 'en')
-                    ->get();
+```php
+$users = Db::table('users')
+                ->where('options->language', 'en')
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:506]:
```suggestion
    ->where('options->language', 'en')
    ->get();
```

File: database-query.md [hunk 13]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
 
-    $users = Db::table('users')
-                    ->where('preferences->dining->meal', 'salad')
-                    ->get();
+$users = Db::table('users')
+                ->where('preferences->dining->meal', 'salad')
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:510]:
```suggestion
    ->where('preferences->dining->meal', 'salad')
    ->get();
```

File: database-query.md [hunk 14]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
 
 You may use `whereJsonContains` to query JSON arrays (not supported on SQLite):
 
-    $users = Db::table('users')
-                    ->whereJsonContains('options->languages', 'en')
-                    ->get();
+```php
+$users = Db::table('users')
+                ->whereJsonContains('options->languages', 'en')
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:518]:
```suggestion
    ->whereJsonContains('options->languages', 'en')
    ->get();
```

File: database-query.md [hunk 15]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
 
 MySQL and PostgreSQL support `whereJsonContains` with multiple values:
 
-    $users = Db::table('users')
-                    ->whereJsonContains('options->languages', ['en', 'de'])
-                    ->get();
+```php
+$users = Db::table('users')
+                ->whereJsonContains('options->languages', ['en', 'de'])
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:526]:
```suggestion
    ->whereJsonContains('options->languages', ['en', 'de'])
    ->get();
```

File: database-query.md [hunk 16]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
 
 You may use `whereJsonLength` to query JSON arrays by their length:
 
-    $users = Db::table('users')
-                    ->whereJsonLength('options->languages', 0)
-                    ->get();
+```php
+$users = Db::table('users')
+                ->whereJsonLength('options->languages', 0)
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:534]:
```suggestion
    ->whereJsonLength('options->languages', 0)
    ->get();
```

File: database-query.md [hunk 17]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
 
-    $users = Db::table('users')
-                    ->whereJsonLength('options->languages', '>', 1)
-                    ->get();
+$users = Db::table('users')
+                ->whereJsonLength('options->languages', '>', 1)
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:538]:
```suggestion
    ->whereJsonLength('options->languages', '>', 1)
    ->get();
```

File: database-query.md [hunk 18]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
-    $users = Db::table('users')
-                    ->when($role, function ($query, $role) {
-                        return $query->where('role_id', $role);
-                    })
-                    ->get();
+$users = Db::table('users')
+                ->when($role, function ($query, $role) {
+                    return $query->where('role_id', $role);
+                })
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:553]:
```suggestion
    ->when($role, function ($query, $role) {
        return $query->where('role_id', $role);
    })
    ->get();
```

File: database-query.md [hunk 19]
```diff
@@ -381,96 +458,116 @@ The `whereNotNull` method verifies that the column's value is **not** `NULL`:
...
-                        return $query->orderBy('name');
-                    })
-                    ->get();
+$users = Db::table('users')
+                ->when($sortBy, function ($query, $sortBy) {
+                    return $query->orderBy($sortBy);
+                }, function ($query) {
+                    return $query->orderBy('name');
+                })
+                ->get();
```
Comment (LukeTowers) [inline database-query.md:569]:
```suggestion
    ->when($sortBy, function ($query, $sortBy) {
        return $query->orderBy($sortBy);
    }, function ($query) {
        return $query->orderBy('name');
    })
    ->get();
```

File: database-relations.md [hunk 20]
```diff
@@ -722,111 +827,139 @@ Dynamic properties are "lazy loading", meaning they will only load their relatio
...
-    }
+foreach ($books as $book) {
+    echo $book->author->name;
+}
+```
 
 For this operation only two queries will be executed:
 
-    select * from books
+```
```
Comment (LukeTowers) [inline database-relations.md:896]:
```suggestion
```sql
```

File: database-structure.md [hunk 21]
```diff
@@ -254,81 +289,95 @@ Command  | Description
...
 
-    $table->foreign('user_id')
-          ->references('id')
-          ->on('users')
-          ->onDelete('cascade');
+```php
+$table->foreign('user_id')
+        ->references('id')
+        ->on('users')
+        ->onDelete('cascade');
```
Comment (LukeTowers) [inline database-structure.md:314]:
```suggestion
    ->references('id')
    ->on('users')
    ->onDelete('cascade');
```

**Review by LukeTowers — CHANGES_REQUESTED** [review]

**Review by LukeTowers — APPROVED** [review]

**Comment by LukeTowers** [comment]
Thanks for all your work on this @WebVPF!

**Pull request merged**